multicluster/controllers/multicluster/core/remote_common_area.go

```diff
@@ -0,0 +1,350 @@
+
+	crtData, token, err := getSecretCACrtAndToken(secret)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create manager for the RemoteCommonArea
+	klog.InfoS("Connecting to RemoteCommonArea", "Cluster", clusterID, "url", url)
+	config, err := clientcmd.BuildConfigFromFlags(url, "")
+	if err != nil {
+		return nil, err
+	}
+	config.BearerToken = string(token)
+	config.CAData = crtData
+	mgr, err := ctrl.NewManager(config, ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		Namespace:          clusterSetNamespace,
+	})
+	if err != nil {
+		klog.ErrorS(err, "error creating manager for RemoteCommonArea", "Cluster", clusterID)
+		return nil, err
+	}
+
+	remoteClient, e := client.New(config, client.Options{Scheme: scheme})
+	if e != nil {
+		return nil, e
+	}
+
+	remote := &remoteCommonArea{
+		Client:                  remoteClient,
+		ClusterManager:          mgr,
+		ClusterSetID:            clusterSetID,
+		ClusterID:               clusterID,
+		config:                  config,
+		scheme:                  scheme,
+		Namespace:               clusterSetNamespace,
+		connected:               false,
+		localClusterClient:      localClusterClient,
+		remoteCommonAreaManager: remoteCommonAreaManager,
+	}
+
+	remoteCommonAreaManager.AddRemoteCommonArea(remote)
+
+	return remote, nil
+}
+
+/**
+ * getSecretCACrtAndToken returns the access credentials from Secret.
+ */
+func getSecretCACrtAndToken(secretObj *v1.Secret) ([]byte, []byte, error) {
+	caData, found := secretObj.Data[v1.ServiceAccountRootCAKey]
+	if !found {
+		return nil, nil, fmt.Errorf("ca.crt data not found in Secret %v", secretObj.GetName())
+	}
+
+	token, found := secretObj.Data[v1.ServiceAccountTokenKey]
+	if !found {
+		return nil, nil, fmt.Errorf("token not found in Secret %v", secretObj.GetName())
+	}
+
+	return caData, token, nil
+}
+
+func (r *remoteCommonArea) SendMemberAnnounce() error {
+	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
+	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+		return err
+	}
+	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
+	localClusterMemberAnnounceExist := false
+	if len(memberAnnounceList.Items) != 0 {
+		for _, memberAnnounce := range memberAnnounceList.Items {
+			if memberAnnounce.ClusterID == string(r.remoteCommonAreaManager.GetLocalClusterID()) {
+				localClusterMemberAnnounceExist = true
+				localClusterMemberAnnounce = memberAnnounce
+				break
+			}
+		}
+	}
+	if localClusterMemberAnnounceExist {
+		if r.remoteCommonAreaManager.GetElectedLeaderClusterID() != common.InvalidClusterID {
+			localClusterMemberAnnounce.LeaderClusterID = string(r.remoteCommonAreaManager.GetElectedLeaderClusterID())
+		}
+
+		if localClusterMemberAnnounce.Annotations == nil {
+			localClusterMemberAnnounce.Annotations = make(map[string]string)
+		}
+		localClusterMemberAnnounce.Annotations["touch-ts"] = time.Now().String()
```

Comment:
Could we define a constant for it? And do you assume time sync between clusters? I feel we should not require that.

Reply:
No time sync needed, this is mainly to force an update on the resource because otherwise the update is a no-op and the reconciler is not invoked

Reply:
Ok, when you define the constant, probably add some comments to explain this.